backend/compact-connect/common_constructs/slack_channel_configuration.py

```diff
@@ -37,7 +37,7 @@ def _configure_chatbot_role(self):
             suppressions=[
                 {
                     'id': 'AwsSolutions-IAM4',
-                    'appliesTo': ['Policy::arn:<AWS::Partition>:iam::aws:policy/job-function/ViewOnlyAccess'],
+                    'applies_to': ['Policy::arn:<AWS::Partition>:iam::aws:policy/job-function/ViewOnlyAccess'],
```

Comment:
I assume this was a typo? What were the implications of this typos, if any? (Genuine question because I don't understand AWS well enough to know. I realize this is a suppression, but I imagine an equivalent typo somewhere else could open a security hole.)

Reply:
Yes, this was a typo I noticed while looking things over. The way these suppressions work is to basically 'turn off' any findings from the rule sets we are using that match the criteria of the suppression. In this case the intended suppression was findings:
- for the resource provided in the first argument
- for the rule called [AwsSolutions-IAM4](https://github.com/cdklabs/cdk-nag/blob/main/RULES.md#errors)
- that are specifically referencing this `ViewOnlyAccess` policy in the `applies_to` field.

The `applies_to` field is a further filter criteria that allows us to be very specific in what we are suppressing. By accidentally miss-naming the field, I was effectively making the suppression broader than I intended because that additional filter criteria was missed. Fixing that typo adds the criteria back in the more specific way I had intended.

Reply:
Got it! I trust you're right, but just confirming...it looks like the docs actually _do_ use `appliesTo`? https://github.com/cdklabs/cdk-nag

Reply:
Yeah, that's the fun of using polyglot libraries (and why I missed the typo in the first place). The library and its docs are written for TypeScript, but, when they get built for their target languages, the interface is updated to match the conventions of the target language (so, `snake_case` for Python).